### ISSUE TITLE
Prince 13-produced PDF has all links removed with obj missing warning

### ISSUE THREAD

**Issue by domenic** [pts/pdfsizeopt#145]
When I run [print-prince13-unoptimized.pdf](https://github.com/pts/pdfsizeopt/files/5284714/print-prince13-unoptimized.pdf) through pdfsizeopt (with default options), I get out 
[print-prince13-optimized-broken.pdf](https://github.com/pts/pdfsizeopt/files/5284716/print-prince13-optimized-broken.pdf), plus hundreds of warnings of the form

```
warning: obj 83823 missing, referenced by objs [968]...
warning: obj 73413 missing, referenced by objs [854]...
warning: obj 19294 missing, referenced by objs [274]...
```

The broken PDF has all of its internal hyperlinks missing. Possibly other things are broken, but that's the most noticeable.

Also interesting: this PDF is produced using Prince from https://html.spec.whatwg.org/. When I use Prince 11 on that same site, I get [print-prince11-unoptimized.pdf](https://github.com/pts/pdfsizeopt/files/5284780/print-prince11-unoptimized.pdf). pdfsizeopt has no problems optimizing that version; the result is 
[print-prince11-optimized.pdf](https://github.com/pts/pdfsizeopt/files/5284725/print-prince11-optimized.pdf).

**Comment by pts** [comment]
Thank you for reporting this! Indeed this looks like a bug in pdfsizeopt. The input file print-prince13-unoptimized.pdf is a hybrid-reference file, meaning that its trailer contain and `/XRefStm` entry. pdfsizeopt currently doesn't support such PDF files. I'll keep the bug open to track adding support.

**Comment by pts** [comment]
Fixed in 7cbb4643b91682bbad2f70eac9220ae1d279285d.
